django/utils/translation/trans_real.py

```diff
@@ -200,6 +200,8 @@ def _add_fallback(self, localedirs=None):
     def merge(self, other):
         """Merge another translation into this catalog."""
         if not getattr(other, '_catalog', None):
+            if self.language() != 'fr':
```

Comment:
If this condition isn't placed, another test `i18n.tests.TranslationTests.test_plural` [1] fails. But will definitely add a comment on why we have that check.

[1] https://github.com/django/django/blob/master/tests/i18n/tests.py#L63